apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetReader.java

```diff
@@ -20,87 +20,91 @@
-
-    }
-
-    public Page<ValueSet> searchValueSearch(ValueSetSearchRequest search, Pageable pageable) {
-        Page<Codeset> rawResults = valueSetRepository.findByCodeSetNmOrValueSetNmorValueSetCodeorValueSetType(search.getCodeSetName(),
-                search.getValueSetNm(), search.getValueSetCode(), search.getValueSetTypeCd(), search.getValueSetDescription(), pageable);
-        List<ValueSet> formatResults = toReadValueSet(rawResults);
-        return new PageImpl<>(formatResults, pageable, rawResults.getTotalElements());
+  private final ValueSetRepository valueSetRepository;
+
+  private final CodeValueGeneralRepository codeValueGeneralRepository;
+
+  ValueSetReader(final ValueSetRepository valueSetRepository,
+      final CodeValueGeneralRepository codeValueGeneralRepository) {
+    this.valueSetRepository = valueSetRepository;
+    this.codeValueGeneralRepository = codeValueGeneralRepository;
+  }
+
+  public Page<ValueSet> findAllValueSets(Pageable pageable) {
+    Page<Codeset> rawResults = valueSetRepository.findAll(pageable);
+    List<ValueSet> formatResults = toReadValueSet(rawResults);
+    return new PageImpl<>(formatResults, pageable, rawResults.getTotalElements());
+
+  }
+
+  public Page<ValueSet> searchValueSet(ValueSetSearchRequest search, Pageable pageable) {
+    Page<Codeset> rawResults = valueSetRepository.findByValueSetNmOrValueSetCodeOrValueSetDescription(
+        search.getValueSetNm(), search.getValueSetCode(), search.getValueSetDescription(), pageable);
+    List<ValueSet> formatResults = toReadValueSet(rawResults);
+    return new PageImpl<>(formatResults, pageable, rawResults.getTotalElements());
+
+  }
+
+  public List<ValueSet> toReadValueSet(Page<Codeset> rawResults) {
+    List<ValueSet> results = new ArrayList<>();
+
+    for (Codeset codeSet : rawResults.getContent()) {
+
+      results.add(new ValueSet(
+          codeSet.getId().getClassCd(),
+          codeSet.getId().getCodeSetNm(),
+          codeSet.getAssigningAuthorityCd(),
+          codeSet.getAssigningAuthorityDescTxt(),
+          codeSet.getCodeSetDescTxt(),
+          codeSet.getEffectiveFromTime(),
+          codeSet.getEffectiveToTime(),
+          codeSet.getIsModifiableInd(),
+          codeSet.getNbsUid(),
+          codeSet.getSourceVersionTxt(),
+          codeSet.getSourceDomainNm(),
+          codeSet.getStatusCd(),
+          codeSet.getStatusToTime(),
+          (codeSet.getCodeSetGroup() != null) ? codeSet.getCodeSetGroup().getId() : null,
+          codeSet.getAdminComments(),
+          codeSet.getValueSetNm(),
+          codeSet.getLdfPicklistIndCd(),
+          codeSet.getValueSetCode(),
+          codeSet.getValueSetTypeCd(),
+          codeSet.getValueSetOid(),
+          codeSet.getValueSetStatusCd(),
+          codeSet.getValueSetStatusTime(),
+          codeSet.getParentIsCd(),
+          codeSet.getAddTime(),
+          codeSet.getAddUserId()));
```

Comment:
Suggestion: Instead of returning all of this can we check to see what fields the UI will need? I expect that we can cut several of these fields.